aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/BaseHandlerStd.java

```diff
@@ -610,17 +581,44 @@ protected void removeOldTags(
         return progress;
     }
 
-    protected <K, V> Map<K, V> mergeMaps(Collection<Map<K, V>> maps) {
-        final Map<K, V> result = new HashMap<>();
-        for (Map<K, V> map : maps) {
-            if (map != null) {
-                result.putAll(map);
-            }
+    protected ProgressEvent<ResourceModel, CallbackContext> updateTags(
+            final AmazonWebServicesClientProxy proxy,
+            final ProxyClient<RdsClient> rdsProxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress,
+            final Tagging.TagSet previousTags,
+            final Tagging.TagSet desiredTags
+    ) {
+        final Tagging.TagSet tagsToAdd = Tagging.exclude(desiredTags, previousTags);
+        final Tagging.TagSet tagsToRemove = Tagging.exclude(previousTags, desiredTags);
+
+        if (tagsToAdd.isEmpty() && tagsToRemove.isEmpty()) {
+            return progress;
+        }
+
+        DBInstance dbInstance;
+        try {
+            dbInstance = fetchDBInstance(rdsProxyClient, progress.getResourceModel());
+        } catch (Exception exception) {
```

Comment:
Nit: Should we add DBInstance reference to callback context and have a simple call chain step to refresh this object. I think that what callback context is meant for. We will avoid many try/catch blocks.

Reply:
The disadvantage of adding some extra attributes to context is that it will try to preserve them by serializing, hence we should explicitly prevent it from being serialized and then de-serialized. I personally see to be quite cumbersome, but surely this could save us a few Describe- calls.